pkg/registry/core/seed/strategy.go

```diff
@@ -51,6 +51,7 @@ func (s Strategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 	seed := obj.(*core.Seed)
 
 	seed.Generation = 1
+	syncDependencyWatchdogSettings(seed)
```

Comment:
Don't we also need to invoke this function here? https://github.com/gardener/gardener/blob/7f396c5b70d5afc40c439c468b246da84566f2cc/pkg/registry/core/seed/strategy.go#L158-L166

Otherwise, what happens when this is deployed and there are no changes in the `spec` of the `Seed`s? We are afraid that this code is not executed in such case, i.e., the new `weeder`/`prober` fields will remain `nil`?

Reply:
thanks , i didn't think that.